sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -499,8 +500,48 @@ case class TruncateTableCommand(
         val path = new Path(location.get)
         try {
           val fs = path.getFileSystem(hadoopConf)
+          val fileStatus = fs.getFileStatus(path)
+          // Not all fs impl. support these APIs.
+          var optPermission: Option[FsPermission] = None
+          try {
+            optPermission = Some(fileStatus.getPermission())
+          } catch {
+            case NonFatal(e) =>
+          }
+
+          var optAcls: Option[java.util.List[AclEntry]] = None
+          try {
+            optAcls = Some(fs.getAclStatus(path).getEntries)
+          } catch {
+            case NonFatal(e) =>
+          }
+
           fs.delete(path, true)
+
+          // We should keep original permission/acl of the path.
+          // For owner/group, only super-user can set it, for example on HDFS. Because
+          // current user can delete the path, we assume the user/group is correct or not an issue.
           fs.mkdirs(path)
+          optPermission.foreach { permission =>
+            try {
+              fs.setPermission(path, permission)
+            } catch {
+              case NonFatal(e) =>
+                throw new AnalysisException(
```

Comment:
Ur, `AnalysisException` sounds weird.

Reply:
Also, `Exception` is too risky behavior change. (@gatorsmile ).
I guess `warning` might be enough?

Reply:
> Ur, `AnalysisException` looks weird a little. This is not `Analysis` error.

Ah, this follows throwing AnalysisException at the outside try..catch block. We can change this.

Reply:
> Also, `Exception` is too risky behavior change. (@gatorsmile ).
> I guess `warning` might be enough?

For security reason, cannot set back original Acl/permission seems serious issue so `Exception` is thrown now. `warning` might be ignored and not noticed by users.

I'm ok with a `warning` if you all think it isn't an issue.

Reply:
> Ur, `AnalysisException` looks weird a little. This is not `Analysis` error.

Maybe `SparkException` or `RuntimeException`?

Reply:
- If this is for security, what about `SecurityException`?
- If we need to raise an exception, shall we have a fallback configuration?

Reply:
Based on previous discussion around https://github.com/apache/spark/pull/26956#discussion_r363163389, we may not need a config.

`SecurityException` seems ok. cc @cloud-fan

Reply:
SecurityException sounds good